<!--
Usage:
  Rscript -e 'library(rmarkdown); render("MethylReport2.Rmd","pdf_document")'
-->

---
title: "MethylReport"
author: "Shazeda Omar"
date: "11/28/2018"
output: html_document

---

## Purpose

The purpose of this project is to develop a software for the analysis of multiple DNA methylation data obtained from Next Generation Sequencing Technology by inputting, annotating and generating data. The software will be able to evaluate DNA methylation data for visualization, location of methylation sites and coverage.


## Background

The human genome contains essential information that is used to maintain the expression of genes and the regulation of proteins in different organ systems. Underlying mechanisms can directly or indirectly affect this process resulting in gene alterations. Epigenetics is a process that creates alterations in the gene expressions without modifying the DNA sequence1. The DNA methylation mechanism that is crucial for development with an important role in many vital processes including X-chromosome inactivation, and suppression of repetitive element transcription and, when dysregulated, contributes to diseases like cancer2. DNA methylation influences gene expression by affecting the nitrogenous cytosine base and areas where the cytosine follows a guanine base, commonly called CpG islands (CGIs) or short interspersed sequences.

Many tools developed can significantly differ the common genomic pattern by being GC-rich, CpG-rich, and predominantly nonmethylated3. Methylation is regulated by DNA  methyltransferases (Dnmts) which shifts the methyl group from the S-adenyl methionine (SAM) to the fifth carbon cytosine residue forming a 5-methylcytosine4.  The detection of DNA methylation uses bisulfite sequencing in  protocols such as Enhanced Reduced Representation Bisulfite Sequencing (ERRBS)5, Reduced Representation Bisulfite Sequencing (RRBS)6, Whole-Genome Bisulfite sequencing (WGBS) and Methylome Capture Sequencing (targeted methylome sequencing)7. The hypothesis conducted will evaluate the Enhanced Reduced Representation Bisulfite Sequencing (ERRBS) protocol which provides improved coverage of DNA methylation sites across specific regions of the genome compared to other protocols.

Bisulfite sequencing detection is used to identify segments of methylation or nonmethylation in the DNA, it uses sodium bisulfite to find the amination reactions of cytosine and 5-methylcytosine(5mC)5.  Cytosines in single-stranded DNA are converted into uracil and recognized as thymine in PCR amplification, but 5mCs are immune to this conversion and remain as cytosines allowing it to be analyzed from unmethylated cytosines8.  Both the RRBS and ERRBS protocol are similar in that they use bisulfite sequencing and restriction enzymes. The RRBS protocol uses methylation-dependent restriction enzymes (MSREs) in fragments of 500-600bps7 whereas the ERRBS is an advanced protocol derived from RRBS which produces more in-depth methylation coverage of CpG islands (CGIs) or cytosines. The ERRBS protocol uses a restriction enzyme that generates a low molecular fragment weight in library preparation5 and yields approximately ten percent of genomic CpG sites9. This provides enrichment in CpG islands and CpG shores, promoters, exons, introns and intergenic regions9.

The Whole genome bisulfite sequencing (WGBS) is produced by Next Generation Sequencing technology (NGS) that can capture all the cytosine in the genome at single-nucleotide resolution but has several drawbacks amplified with increasing sample numbers7. While the WGBS is increasingly accessible for both primary and clinical research10,  its cost has remained substantial and limits the widespread use for multi-sample comparison of large methylomes such as mammals11. An alternative to WGBS is Methylome Capture Sequencing or targeted methylome sequencing (TMS). Methylome Capture Sequencing is based on the capture of methylated DNA using the methyl-binding domain of methyl CpG binding protein and subsequent NGS of eluted DNA, allowing the DNA to be separated by salt gradients according to its CpG methylation density12. An aligner tool, Bismark uses a combination of bisulfite treatment of DNA and high throughput sequencing (BS-Seq) for efficient time analysis, read mapping and methylation calling13.


```{r, include=FALSE}
# library(rmarkdown)
library(methylKit)
library(GenomicRanges)
library(vioplot)

# (thk2008) this doesn't do anything
# source("/Users/shazedaomar/Desktop/Internship_notes/MethylCall_Data_Epicore")

#set up some variables
analysis.directory  <- "/Users/shazedaomar/Desktop/Internship_notes"
working.directory   <- paste(analysis.directory, "MethylCall_Data_Plots", sep="/")
data.file.directory <- paste(analysis.directory, "MethylCall_Data_Epicore", sep="/")

# (thk2008) for me on my machine
if (Sys.getenv()[["USER"]] == "thk2008") {
  analysis.directory  <- "/scratch001/thk2008/wcm-epicore-internship/errbs-vs-rrbs/Analysis_edits"
  working.directory   <- paste(analysis.directory, "MethylCall_Data_Plots", sep="/")
  data.file.directory <- paste(analysis.directory, "MethylCall_Data_Epicore", sep="/")

  # (thk2008) ive already run this so for testing
  # i want to turn it off
  run_methylation <- FALSE
}

# set the working directory
# (thk2008) i was trying to set the working directory globally for all chunks
# but it's not working, so I'll paste it onto the files
if (file.exists(working.directory)) {
  cat("set working directory to [",working.directory,"]\n")
  knitr::opts_knit$set(root.dir=working.directory)
  setwd(working.directory)
} else {
  stop("Cannot find working directory [",working.directory,"]")
}
```

<!-- (thk2008) this is done below -->
<!--
```{r, include=FALSE}
load("myobj.rda")
```
-->

```{r, include=FALSE}
if  (!exists("myobj")) {    #if not myobj will be created
  cat("Did not find myobj\n")
  rda <- paste(working.directory, "myobj.rda", sep="/")
  if(file.exists(rda)) {
    cat("myobj.rda loading [",rda,"]\n")
    load(rda)
  } else {
    cat("creating and saving myobj\n")
    #read in files - path file stored in variable data.file.path
    file.names.list <- list.files(path = data.file.directory, pattern = ".mincov0.txt.gz", full.names = TRUE)
    myobj <- methRead(as.list(file.names.list),
      sample.id=as.list(gsub(".mincov0.txt.gz", "", basename(file.names.list))), #gsub finds all the files that end in ("*.mincov0.txt.gz"), Basename removes the ending of the file.
      assembly="hg19",
      treatment=rep(0,6), #rep function - 0 = same , 6 = number of items in the file.names.list
      context="CpG",
      dbdir = "methylDB"
    )
    save(myobj,file = rda)
    cat ("myobj saved [",rda,"]\n")
  }
}

```

```{r, include=FALSE}

# (thk2008) generate a pdf for each sample with methylation and coverage stats
# functions could be kept in a separate utility file and sourced at the beginning
methylation <- function(myobj) {
    for (i in 1:length(myobj)) {
        sample.output <- myobj[[i]]@sample.id
        cat("sample", sample.output, "\n", sep=": ")
        cat("number of sites", nrow(myobj[[i]]), "\n", sep=": ")
        pdf(paste(working.directory, paste(sample.output,"methylation_output.pdf",sep="_"), sep="/"))
        getMethylationStats(myobj[[i]])
        getMethylationStats(myobj[[i]],plot=TRUE,both.strands=FALSE)
        getCoverageStats(myobj[[i]],plot=TRUE,both.strands=FALSE)
        dev.off() #pdf saved in current directory
    }
}

methylation(myobj) #to execute function --error!!
```

```{r, echo=FALSE}
#Output number of rows for each sample - cpg sites
samples <- sapply(myobj, nrow)
names(samples) <- getSampleID(myobj)
# List the number of rows (CpGs) in the samples
print(samples)

```

```{r, include = FALSE}
#save meth file so that it doesn't have to be created
# (thk2008) see what the methRead() does by default,
# this may be unnecessary
if  (!exists("filter_myobj")) {
  cat("Did not find filter_myobj\n")
  rda <- paste(working.directory, "filter_myobj.rda", sep="/")
  if(file.exists(rda)) {
    cat("filter_myobj.rda loading [",rda,"]\n")
    load(rda)
  } else {
    cat("creating and saving filtered myobj")
    cat("is this necessary? see ?methRead")
    filter_myobj = filterByCoverage(myobj,lo.count=10,lo.perc=NULL,hi.count=NULL,hi.perc=99.9)
    save(filter_myobj,file = rda)
    cat ("filter_myobj.rda saved [",rda,"]\n")
  }
}

```

```{r, include= FALSE}
#save meth file so that it doesn't have to be created
if  (!exists("meth")) {
  cat("Did not find meth.myobj\n")
  rda <- paste(working.directory, "meth.rda", sep="/")
  if(file.exists(rda)) {
    cat("meth.rda loading [",rda,"]\n")
    load(rda)
  } else {
    cat("creating and saving meth object")
    meth=unite(myobj, destrand=FALSE)
    save(meth,file = rda)
    cat ("meth.rda saved [",rda,"]\n")
  }
}

#load meth.rdathat is already created
# (thk2008) this is done above
#load("meth.rda")
#load("filter_myobj.rda")

```


```{r, echo=FALSE}
perc.meth=percMethylation(meth)
dens <- apply(perc.meth, 2, density)
# (thk2008) fix this graph
plot(NA, xlim=range(sapply(dens, "[", "x")), ylim=range(sapply(dens, "[", "y")))
mapply(lines, dens, col=1:length(dens))
legend("topright", legend=names(dens), fill=1:length(dens))

```


The graph above displays the Percent Methylation for the given samples ERRBS, methylCaptureA, MethylCaptureB, MethylCaptureC, RRBS and WGBS. The six files have been merged into one object for comparison.



for loop and function to obtain correlation and Cluster Samples

```{r}

# (thk2008) what is this doing?
# what do you want it to do?
Analysis <- function(meth) {

    for (i in 1:length(meth)) {
        getCorrelation(meth,plot=TRUE)
        clusterSamples(meth, dist="correlation", method="ward", plot=TRUE) #find similarity in method
        #Principal component analysis - brings out variation and strong pattern in sample (helps to visualize/explore)
        PCASamples(meth, screeplot=TRUE)
        PCASamples(meth) #is this important

        break
    }
}

# (thk2008) do not run this until it's fixed
# Analysis(meth)

```


```{r}

# (thk2008) the graphs need some fixing and 
# the device logic needs clarification
stop("do not run this until the logic is fixed")

#plots and graphs - 1) barplot of all samples
barplot(samples,
main="Analysis",
xlab="Methylation Method",
ylab="Frequency",
col= "blue")
dev.new()
pdf(paste(barplot(samples), "barplot_samples.pdf", sep = "_"))
#dev.off()

# 2) plot the samples
# (thk2008) fix this plot
plot(samples,
main="Analysis",
main = "",
xlab = "" ,
ylab = "frequency",
type = "o", #plot type
col = "red") #plot color
axis(1, labels=names(filt_coverage_vals), at=seq(1,6), las=2) #changes the labels 1,2,3,4,5 with sample.id names (works with this line of code)
dev.new()
pdf(paste(plot(samples), "plot_samples.pdf", sep = "_"))
#dev.off()
```



