# Schedule

## Background
- ERRBS and RRBS libraries
- What's the difference in the library preparation protocols?
  - The [ERRBS paper](http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1002781)
- what about other methods? For example wgbs, methylome capture

## Methylcall data
- We use [Bismark](https://www.bioinformatics.babraham.ac.uk/projects/bismark/) to align the sequencing reads (fastq) and genreate methylation calls
  - It would be useful to look at [Bismark paper](https://academic.oup.com/bioinformatics/article/27/11/1571/216956) and their [RRBS guide](https://github.com/FelixKrueger/TrimGalore/blob/master/Docs/RRBS_Guide.pdf)
  - The authors of the errbs paper developed, [methylKit](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3491415/), software for analyzing methyllation data

## Find the datasets for each

## Description of Project
- in R or Python
  - read in methylcall files
  - plot methylation values
  - compare site calls
    - how man sites
    - how many common
  
